docs/conf.py

```diff
@@ -285,7 +287,67 @@
 }
 
 
-# Options for sphinx-hoverxref options
-# ------------------------------------
+# Options for the sphinx-hoverxref extension
+# ------------------------------------------
 
 hoverxref_auto_ref = True
+
+
+# Options for the linkcode extension
+# ----------------------------------
+
+# Based on numpy’s:
+# https://github.com/numpy/numpy/blob/dedc4178fc334329de9872ab42df870d2ac7a270/doc/source/conf.py#L313
+def linkcode_resolve(domain, info):
```

Comment:
Hey! My only concern is that we get a large untested function in a config file. Is it possible to move it somewhere else, and add some tests for it (at least doctests)?

Reply:
Moved into a separate extension, implementation simplified for our use case, and tests added.